src/krb5/_localauth.pyx

```diff
@@ -0,0 +1,47 @@
+# Copyright: (c) 2022 Jordan Borean (@jborean93) <jborean93@gmail.com>
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+
+from libc.stdlib cimport free, malloc
+
+from krb5._exceptions import Krb5Error
+
+from krb5._context cimport Context
+from krb5._krb5_types cimport *
+from krb5._principal cimport Principal
+
+
+cdef extern from "python_krb5.h":
+
+    krb5_error_code krb5_aname_to_localname(
+        krb5_context context,
+        krb5_const_principal aname,
+        int lnsize,
```

Comment:
Looking at Heimdal, this is declared as `size_t` https://github.com/heimdal/heimdal/blob/fa92fe37e7ed1cfbe546fd617d712f47c55e35b9/lib/krb5/aname_to_localname.c#L299. I'm no expert at C or how the ABIs work but I would have thought this could potentially cause some problems.

It might require some trickery that we do in [_keyblock.pyx](https://github.com/jborean93/pykrb5/blob/f36e82ffa4ddd9f540a0bdf41c13a2d404df284d/src/krb5/_keyblock.pyx#L15) to define a generic function but use compiler directives to cast as needed to `size_t` on Heimdal.